Implement `check_icdf` helper to test icdf implementations

ricardoV94 opened the pull request:
For use in #6528 

Also cleaned up the `check_logp` like helpers

CC @michaelraczycki

ricardoV94 committed:
Add flag to `CheckParameterValue` to inform whether it can be replaced by -inf

michaelosthege commented on pymc/testing.py:
```diff
@@ -541,6 +542,110 @@ def check_logcdf(
+    decimal : Int
+        Level of precision with which pymc_dist and scipy_icdf are compared.
+        Defaults to 6 for float64 and 3 for float32
+    n_samples : Int
```
```suggestion
    decimal: Optional[int]=None,
    n_samples: int=100,
):
    """
    Generic test for PyMC icdf methods

    The following tests are performed by default:
        1. Test PyMC icdf and equivalent scipy icdf (ppf) methods give similar
        results for parameters inside the supported edges.
        Edges are excluded by default, but can be artificially included by
        creating a domain with repeated values (e.g., `Domain([0, 0, .5, 1, 1]`)
        2. Test PyMC icdf method raises for invalid parameter values
        outside the supported edges.
        3. Test PyMC icdf method returns np.nan for values below 0 or above 1,
         when using valid parameters.

    Parameters
    ----------
    pymc_dist: PyMC distribution
    paramdomains : Dictionary of Parameter : Domain pairs
        Supported domains of distribution parameters
    scipy_icdf : Scipy icdf method
        Scipy icdf (ppp) method of equivalent pymc_dist distribution
    decimal : int, optional
        Level of precision with which pymc_dist and scipy_icdf are compared.
        Defaults to 6 for float64 and 3 for float32
    n_samples : int
```

ricardoV94 replied:
Done

ricardoV94 committed:
Simplify `check_logp` and related testing helpers

ricardoV94 committed:
Implement `check_icdf` helper to test icdf implementations

Note that adding a nan switch to the icdf expression of discrete variables, prevents the returned dtype to be the same as the original distribution. There is no integer nan!

ricardoV94 committed:
Do not test invalid parameter combinations

michaelosthege reviewed (commented):
LGTM
